hstream/src/HStream/Server/Types.hs

```diff
@@ -155,3 +163,21 @@ printAckedRanges :: Map.Map ShardRecordId ShardRecordIdRange -> String
 printAckedRanges mp = show (Map.elems mp)
 
 type ConsumerName = T.Text
+
+--------------------------------------------------------------------------------
+-- shard
+
+getShardName :: Int -> CB.CBytes
+getShardName idx = textToCBytes $ "shard" <> T.pack (show idx)
+
+getShard :: HS.LDClient -> HS.StreamId -> Maybe T.Text -> IO HS.C_LogID
+getShard client streamId key = do
+  partitions <- HS.listStreamPartitions client streamId
+  let size = length partitions - 1
+  let shard = getShardName . getShardIdx size <$> key
+  HS.getUnderlyingLogId client streamId shard
+
+getShardIdx :: Int -> T.Text -> Int
+getShardIdx size key = let hashValue = hash key
+                        in hashValue `mod` size
+
```

Comment:
Consider putting this information into HStream.Store

Reply:
I think it's best to wait until the entire shard-related algorithm is implemented before considering whether migration is needed.